src/internal_modules/roc_pipeline/receiver_session_group.cpp

```diff
@@ -188,12 +190,44 @@ void ReceiverSessionGroup::route_control_packet_(const packet::PacketPtr& packet
 bool ReceiverSessionGroup::can_create_session_(const packet::PacketPtr& packet) {
     if (packet->flags() & packet::Packet::FlagRepair) {
         roc_log(LogDebug, "session group: ignoring repair packet for unknown session");
+        drop_packet_(packet);
         return false;
     }
 
     return true;
 }
 
+void ReceiverSessionGroup::drop_packet_(const packet::PacketPtr& packet_ptr) {
+    dropped_packet_buffer_.push_back(*packet_ptr);
+
+    const size_t max_size = receiver_config_.common.max_session_packets
+        * receiver_config_.default_session.max_sessions;
+
+    if (max_size == 0) {
+        return;
+    }
```

Comment:
This is actually a dangerous behavior. It means that if max_sessions or max_session_packets is zero, we can accumulate pre-buffered packets indefinitely. If user hasn't set one of the limits, and some peer sends us some ill-formed packets (intentionally or not), and we never create session for those packets, we will consume more and more memory until we eat everything available on system.

Given it second thought, it seems that using `max_session_packets * max_sessions` formula here is not so good idea. **I suggest to change approach a bit:** instead of max_session_packets and max_sessions, introduce `prebuf_len` setting, measured in nanoseconds (core::nanoseconds_t).

Instead of limiting number of packets, it will limit the age of packets. When the age of the oldest (i.e. the first) packet in buffer becomes larger than prebuf_len, that packet is removed from the buffer.

We have a good default value for `prebuf_len`: if the user did not set it, it will be equal to `target_latency` - i.e. we will accumulate as much packets as needed for session start. We can add `prebuf_len` to ReceiverSessionConfig. We also need to add `deduce_prebuf_len()`, that sets prebuf_len to target_latency, and is called if the user did not provide `--prebuf-len` explicitly (similar to `deduce_resampler_backend()`).

Reply:
To compute the age of a packet, we'll need to add **receive timestamps** to packets. Then, the age can be computed as `now - receive_timestamp`.

It should be easy:

* add receive_timestamp field here: https://github.com/roc-streaming/roc-toolkit/blob/develop/src/internal_modules/roc_packet/target_libuv/roc_packet/udp.h

* fill it here: https://github.com/roc-streaming/roc-toolkit/blob/develop/src/internal_modules/roc_netio/target_libuv/roc_netio/udp_receiver_port.cpp